### ISSUE TITLE
Got an error when opening console in a dev build

### ISSUE THREAD

**Issue by tony612** [bitwalker/distillery#25]
When I opened the console after building the dev release(`$ mix release`), I got and error:
```
$ bin/myapp console
[info] Application myapp exited: Myapp.start(:normal, []) returned an error: shutdown: failed to start child: Myapp.Endpoint
    ** (EXIT) shutdown: failed to start child: Phoenix.CodeReloader.Server
        ** (EXIT) an exception was raised:
            ** (UndefinedFunctionError) function Mix.Project.config/0 is undefined (module Mix.Project is not available)
                Mix.Project.config()
                (phoenix) lib/phoenix/code_reloader/server.ex:29: Phoenix.CodeReloader.Server.init/1
                (stdlib) gen_server.erl:328: :gen_server.init_it/6
                (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

It's a Phoenix project with code_reloader on in dev env.

**Comment by akeating** [comment]
try adding `:mix` to your `mix.exs` application list.

**Comment by bitwalker** [comment]
@tony612 @akeating In general I would avoid including `:mix`, but the problem here is that the code reloader is not intended for use within releases. You're better off either building your release with `MIX_ENV=prod mix release --env=dev`, or disabling the code reloader temporarily while doing development against the release.

**Comment by tony612** [comment]
@bitwalker Got it. Thanks!

**Comment by clessg** [comment]
This is a pretty confusing issue to run into when following the Walkthrough's [Building your release](https://hexdocs.pm/distillery/walkthrough.html#building-your-release) section. It tells you to just run `mix release`, and then this error shows up.

**Comment by bitwalker** [comment]
@clessg That's a good point, there are actually Phoenix-specific walkthroughs, but after skimming them, they do not explicitly identify this as something to be aware of. I'll make sure they get updated to mention this.

**Comment by tgodfrey** [comment]
I found that using `mix release --env=prod`, I was still getting the error above. I was getting both `_build/dev`, and `_build/prod`, and the instructions at the end said to run the release from `_build_dev`.

Using `MIX_ENV=prod mix release --env=prod`, I was able to get a clean prod release, which works like a charm.

Distillery is a godsend, BTW. Much easier to use, IMHO, than exrm. Thanks for the hard work on it!

**Comment by mkrogemann** [comment]
Really glad I found this issue before going crazy :) 
I would like to run a build on my dev box in dev mode and disable the code reloader as suggested. I would however like to NOT do this in the config of the Phoenix app though. Instead I wish I could either pass it in on the command line or specify it in the release config. I may be overlooking something here but cannot currently find a way to do this in the various docs I looked up. 
Can anyone here give a hint?
Thanks!
